fpdf/fonts.py

```diff
@@ -438,7 +460,45 @@ def __init__(
     @property
     def hbfont(self):
         if not self._hbfont:
-            self._hbfont = HarfBuzzFont(hb.Face(hb.Blob.from_file_path(self.ttffile)))
+            # HarfBuzz expects an SFNT (TTF/OTF) blob. For safety and for
+            # WOFF/WOFF2 input we re-serialize the fontTools TTFont to a
+            # raw SFNT byte buffer and hand that to HarfBuzz. This avoids
+            # relying on harfbuzz to accept compressed web font containers.
+            try:
+                from io import BytesIO
+
+                buf = BytesIO()
+                # Ensure we have the decompressed tables in memory
+                self.ttfont.save(buf)
+                buf.seek(0)
+                ttfont_bytes = buf.read()
+            except Exception:
+                # Fallback: attempt to let HarfBuzz load from file path directly
+                self._hbfont = HarfBuzzFont(hb.Face(hb.Blob.from_file_path(self.ttffile)))
+                return self._hbfont
+
+            # Try to create a HarfBuzz blob from bytes; if not available, write a
+            # temporary file as a last resort.
+            try:
+                blob = hb.Blob.from_bytes(ttfont_bytes)
+                face = hb.Face(blob)
+            except Exception:
+                import tempfile, os
+
+                tmp = tempfile.NamedTemporaryFile(suffix=".ttf", delete=False)
+                try:
+                    tmp.write(ttfont_bytes)
+                    tmp.flush()
+                    tmp.close()
+                    face = hb.Face(hb.Blob.from_file_path(tmp.name))
+                finally:
+                    try:
+                        os.unlink(tmp.name)
+                    except Exception:
+                        # Best-effort cleanup; if it fails, leave the temp file.
+                        pass
+
+            self._hbfont = HarfBuzzFont(face)
```

Comment:
There is a performance hit when we load non-WOFF fonts with your code since we add an extra step to save the font to a byte array.
I believe keeping the current behavior for non-WOFF fonts, and applying your load/save to byte array/load the byte array on harfbuzz to WOFF/WOFF2 fonts only would be better performance-wise.